### ISSUE TITLE
Is it possible to change mx.symbol.SoftmaxOutput's grad_scale on the fly?

### ISSUE THREAD

**Issue by ufukcbicici** [apache/incubator-mxnet#3745]
For an algorithm I am developing, I need to scale the gradients starting backpropagating from the SoftmaxOutput units by a scalar, which changes in every iteration of the training. Since (I believe), SoftmaxOutput does not accept gradients from below layers (as it is meant to be the last operation), calling backward() with a proper out_grads parameter does not work. Therefore currently in each iteration I connect a new mx.symbol.SoftmaxOutput to the last feature layer, with the newly calculated scalar for that iteration in the grad_scale parameter and then bind a new executor object.

Is there any way to change grad_scale parameter of the SoftmaxOutput, once the executor has been bound? I obvserved that binding a new executor in each iteration with the way I explained above introduces a significant amount of overhead. What would be the best way to that instead?

Thanks in advance.

**Comment by piiswrong** [comment]
SoftmaxOutput(out_grad=True)
then you can change individual scale for each instance

**Comment by ufukcbicici** [comment]
Thanks for the answer. I add the softmax output as:

            softmax_output = mx.symbol.SoftmaxOutput(data=self.leafNodePreOutputLayer, name=softmax_object_name, out_grad=True)


And then call the backward pass on the executor I bind to that network as:

            grad_scale = self.get_gradient_scale()
            grad_array = self.outputIndexToArgumentDict[0].gradientArray
            grad_array[:] = grad_scale
            self.executor.backward(out_grads=[grad_array])


Still, I am seeing the same unscaled derivatives. Somehow it seems that the operator does not get gradients despite setting out_grad=True. Can this be a bug?

**Comment by ufukcbicici** [comment]
Ok, I think the reason is that I am using an old mxnet.dll. The softmax_output-inl.h file in the source code I have compiled does not declare out_grad as a parameter. I will compile with the latest source code and check again.

**Comment by yajiedesign** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
